### PR TITLE
Add Multiplex signal support. Fix big endian decoding.

### PR DESCRIPTION
**Pull request by agent6262**
Fix for #44, #45 and #13. However this change is not fully featured as when adding and removing signals to the dbc they do not properly reflect in the object (also writing the db might not show multiplexing correctly?)

### CODE REVIEW

**Comment by agent6262** [comment]
@bit-dream as in my pr comment I was not able to completely finish adding proper multiplex support. Would definitely need help on dbc adding / removing and dbc writing.

**Review by bit-dream — CHANGES_REQUESTED** [review]
Hey, sorry for getting back to you late on this.

Overall, the changes look great. However, I think there are still some issues with the decode implementation, I don't think it's quite there yet/accurately decodes signals. I tried to be as detailed as I could with the comments, so check those out and let me know what you think.

That said, I would be willing to merge all of your changes into main, except for the changes to Can and BitUtils. If you want to want to open up a separate pull request for those other changes, I'll get those added, as I like what you did there.

As for the signal decoding logic, we may just need to revisit that. If you want me to take a stab at it I am happy too. If you want to take that on, you are welcome to as well.

**Comment by bit-dream** [comment]
> @bit-dream as in my pr comment I was not able to completely finish adding proper multiplex support. Would definitely need help on dbc adding / removing and dbc writing.

That shouldn't be an issue at all. I can work on adding that functionality.

File: src/can/BitUtils.ts [hunk 1]
```diff
@@ -1,6 +1,17 @@
 import { EndianType } from '../shared/DataTypes';
 
 class BitUtils {
+  static bigEndianBitOrder = [
```
[the remark this answers is not in the document]
Comment (agent6262) [reply src/can/BitUtils.ts]: I wont even attempt to claim to properly understand how can and candfd bit ordering stuff works. I am unable to find good documentation on it, so I was just attempting solutions till it got what I expected.

But as far as can fd goes I not super worried about it right now, as this lib does not account for it yet, but when that feature is added checks and function will need to change for it.

**Comment by agent6262** [comment]
@bit-dream I removed the can and bit changes as requested.

**Review by bit-dream — APPROVED** [review]

**Comment by bit-dream** [comment]
@agent6262 I approved your pull request and merged into main. Awesome work and thanks for contributing.

I'll take a stab at implementing a more robust decode method. I'll update issue #45 so you can be in the loop when those changes get pulled in.

Additionally, if you are looking to be an active maintainer, feel free to give me a shout.

**Pull request merged**